### PR TITLE
GH-179 Knockback rework

### PR DESCRIPTION
**Pull request by CitralFlo**
Fix

### CODE REVIEW

File: eternalcombat-api/src/main/java/com/eternalcode/combat/region/RegionController.java [hunk 1]
```diff
@@ -17,12 +17,15 @@ public class RegionController implements Listener {
...
     private final PluginConfig pluginConfig;
+    private final double knockbackMultiplier;
 
     public RegionController(NotificationAnnouncer announcer, RegionProvider regionProvider, FightManager fightManager, PluginConfig pluginConfig) {
         this.announcer = announcer;
         this.regionProvider = regionProvider;
         this.fightManager = fightManager;
         this.pluginConfig = pluginConfig;
+
+        this.knockbackMultiplier = this.pluginConfig.settings.blockedRegionKnockMultiplier;
```
Comment (Rollczi) [inline eternalcombat-api/src/main/java/com/eternalcode/combat/region/RegionController.java:28]: You should use variable directly from plugin  config.

File: eternalcombat-api/src/main/java/com/eternalcode/combat/notification/NotificationAnnouncer.java [hunk 2]
```diff
@@ -69,6 +74,26 @@ public void broadcast(String text) {
...
+
+        UUID uniqueId = player.getUniqueId();
+
+        if (this.timeLimitMap.containsKey(uniqueId)) {
+            Instant instant = this.timeLimitMap.get(uniqueId);
+
+            if (instant.isAfter(Instant.now())) {
+                return;
+            }
+            else {
```
Comment (P1otrulla) [inline eternalcombat-api/src/main/java/com/eternalcode/combat/notification/NotificationAnnouncer.java:89]: Not needed else block

File: eternalcombat-api/src/main/java/com/eternalcode/combat/notification/NotificationAnnouncer.java [hunk 3]
```diff
@@ -15,6 +19,7 @@ public final class NotificationAnnouncer {
 
     private final AudienceProvider audienceProvider;
     private final MiniMessage miniMessage;
+    private final HashMap<UUID, Instant> timeLimitMap = new HashMap<>();
```
Comment (P1otrulla) [inline eternalcombat-api/src/main/java/com/eternalcode/combat/notification/NotificationAnnouncer.java:22]: Im not sure is good place for holding this

File: eternalcombat-api/src/main/java/com/eternalcode/combat/notification/NotificationAnnouncer.java [hunk 4]
```diff
@@ -69,6 +74,26 @@ public void broadcast(String text) {
         audience.sendMessage(message);
     }
 
+    public void sendMessage(Player player, String text, Duration timeLimit) {
```
Comment (P1otrulla) [inline eternalcombat-api/src/main/java/com/eternalcode/combat/notification/NotificationAnnouncer.java:77]: You can just copypaste from EternalCore  Delay class end hold this in listener or specific manager for that

File: eternalcombat-api/src/main/java/com/eternalcode/combat/notification/NotificationAnnouncer.java (hunk 3 above)
Comment (imDMK) [inline eternalcombat-api/src/main/java/com/eternalcode/combat/notification/NotificationAnnouncer.java:22]: This should be in a different class. Maybe RegionKnockbackService

File: eternalcombat-api/src/main/java/com/eternalcode/combat/notification/NotificationAnnouncer.java (hunk 4 above)
Comment (imDMK) [inline eternalcombat-api/src/main/java/com/eternalcode/combat/notification/NotificationAnnouncer.java:77]: ^

File: eternalcombat-api/src/main/java/com/eternalcode/combat/region/WorldGuardRegionProvider.java [hunk 5]
```diff
@@ -30,33 +30,26 @@ public Location getRegionCenter(Location location) {
         RegionQuery regionQuery = regionContainer.createQuery();
         ApplicableRegionSet applicableRegions = regionQuery.getApplicableRegions(BukkitAdapter.adapt(location));
 
-        double minX = 0;
-        double maxX = 0;
-        double minZ = 0;
-        double maxZ = 0;
+        ProtectedRegion combatRegion = null;
```
Comment (imDMK) [inline eternalcombat-api/src/main/java/com/eternalcode/combat/region/WorldGuardRegionProvider.java:33]: Maybe use AtomicReference?
```suggestion
        AtomicReference<ProtectedRegion> combatRegion = new AtomicReference<ProtectedRegion>();
```

**Review by imDMK — CHANGES_REQUESTED** [review]

**Pull request closed without merging**